src/HCSR04.cpp

```diff
@@ -7,34 +7,50 @@
 #include "HCSR04.h"
 
 UltraSonicDistanceSensor::UltraSonicDistanceSensor(
-        int triggerPin, int echoPin) {
+        int triggerPin, int echoPin, int maxDistanceCm) {
     this->triggerPin = triggerPin;
     this->echoPin = echoPin;
+    this->maxDistanceCm = maxDistanceCm;
+    this->timeoutMicroSec = 0;
     pinMode(triggerPin, OUTPUT);
     pinMode(echoPin, INPUT);
 }
 
+void UltraSonicDistanceSensor::setTimeoutMicroSec(uint32_t timeoutMicroSec) {
```

Comment:
Instead of this being a method, could we make it a parameter of constructor?
I prefer defining objects once and having them be immutable, then being able to change their behavior on the fly, unless it is really needed for a specific use case.

Reply:
Ok for me, commit pushed.